supabase/migrations/20250503202924_cashu-send-swaps.sql

```diff
@@ -154,6 +155,20 @@ begin
 
     v_transaction_id := v_swap.transaction_id;
 
+    -- Check if there's a reversal transaction pointing to this transaction
+    select state into v_reversal_transaction_state
+    from wallet.transactions
+    where reversed_transaction_id = v_transaction_id;
+
+    if v_reversal_transaction_state is not null then
+        -- If there's a reversal transaction that is not failed, return early
+        -- The token swap completion will handle updating the send swap state
+        if v_reversal_transaction_state != 'FAILED' then
```

Comment:
shouldn't we always return here? why would you proceed with this function if the state is failed if `fail_cashu_token_swap` will set it to completed?